doc/manual/packages.rst

```diff
@@ -621,11 +629,22 @@ not domain experts*.
 
   * ``MATLAB.jl`` provides an interface to call the MATLAB engine from within Julia.
 
+6. Creative names are cool, but confusion isn't so much.
+
+  * Many modern programming languages such as Ruby and Python have popular packages
+    that have entertaining names. Python's ``pickle`` package isn't about pickles, it's
+    for serializing (or pickling) object structures. Ruby's ``Sinatra`` package is named
+    after the popular American singer and is a popular web framework that competes with Ruby's
+    ``Rails`` web framework (another creative name).
+
+  * If being a creative package namer isn't your style, it's okay to not force it.
```

Comment:
Well, `CoffeeScript` *is* descriptive, as are variants with `Mocha`, `Latte`, or whatever your favorite is... one doesn't *have* to be boring just to be descriptive.  `pickle`, `cucumber`, and `Pandas` sound cute, but also are a royal pain if you are trying to Google search for the computer term!
Maybe your section could elaborate a bit on that, providing descriptive, but not necessarily boring names (but saying that boring is fine also, as long as it is descriptive).